### ISSUE TITLE
failed `amplify update api` partially changes resource's auth type

### ISSUE THREAD

**Issue by josefaidt** [aws-amplify/amplify-category-api#565]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

pnpm

### If applicable, what version of Node.js are you using?

18.3.0

### Amplify CLI Version

8.5.1

### What operating system are you using?

mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

no

### Amplify Categories

auth, api

### Amplify Commands

add, init, update

### Describe the bug

see repro steps

### Expected behavior

Updating the API configuration should not be (partially) blocked by a compilation error

### Reproduction steps

1. `amplify init -y`
2. `amplify add api` > GraphQL > accept defaults (default auth is api key) > sample Todo schema
3. add an API Key auth rule to Todo model
	```graphql
    type Todo @model @auth(rules: [{ allow: public }]) {
      id: ID!
      name: String!
      description: String
    }
	```
4. realize you chose the wrong default auth mechanism, run `amplify update api` to change
5. choose Cognito User Pools
6. observe error, but we're unclear if the resource was or was not successfully updated as some config has changed:
	```
	🛑 @auth directive with 'apiKey' provider found, but the project has no API Key authentication provider configured.
	```
7. change auth rules to `allow: owner`
8. upon pushing we receive `Parameters: [AuthCognitoUserPoolId] must have values`

**cli-inputs.json**
```json
{
  "version": 1,
  "serviceConfiguration": {
    "apiName": "gqltest",
    "serviceName": "AppSync",
    "gqlSchemaPath": "/Users/josef/Documents/projects/aws-amplify/reproductions/gqltest/amplify/backend/api/gqltest/schema.graphql",
    "defaultAuthType": {
      "mode": "AMAZON_COGNITO_USER_POOLS",
      "cognitoUserPoolId": "authgqltest7785d8db"
    },
    "conflictResolution": {},
    "additionalAuthTypes": []
  }
}
```

**backend-config.json**
```json
"api": {
    "gqltest": {
      "service": "AppSync",
      "providerPlugin": "awscloudformation",
      "dependsOn": [],
      "output": {
        "authConfig": {
          "defaultAuthentication": {
            "authenticationType": "API_KEY",
            "apiKeyConfig": {
              "apiKeyExpirationDays": 7
            }
          },
          "additionalAuthenticationProviders": []
        }
      }
    }
  },
```

after preemptively changing all API Key auth rules to `owner` auth rules and re-running `amplify update api`, we notice the update now proceeds as expected and all configuration is updated.

### GraphQL schema(s)

<details>

```graphql
type Todo @model @auth(rules: [{ allow: public }]) {
  id: ID!
  name: String!
  description: String
}
```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by palpatim** [comment]
Agreed that this is an unclear error message, but we are going to resolve this as a "won't fix" since there is a workaround, and the Gen 2 flow handles this more robustly.
